### ISSUE TITLE
Need to build stable kernel snaps with build-base:core24 or build-base:devel

### ISSUE THREAD

**Issue by xnox** [snapcore/snapcraft#4283]
### Bug Description

I am building stable kernel snaps for 24-hwe track for Mantic.

I have tried using `build-base: core24` and `build-base: devel`.

`build-base: core24` is not recognised, as there is no dictionary key for it. Can it please be mapped to devel?

```
  File "/snap/snapcraft/9362/lib/python3.8/site-packages/snapcraft/parts/lifecycle.py", line 537, in _run_in_provider
    build_base = providers.SNAPCRAFT_BASE_TO_PROVIDER_BASE[snapcraft_base]
KeyError: 'core24'
```

`build-base: devel` requires `grade: devel` meaning I will not be able to promote mantic kernel snaps to 24-hwe/stable, and thus will unable to build Desktop Images using it

### To Reproduce

Use pc-kernel snapcraft.yaml from jammy, try building it with `build-base: core24` or `build-base: devel` whilst having grade set to stable.

### Environment

Ubuntu

### snapcraft.yaml

```shell
name: pc-kernel
adopt-info: kernel
grade: devel
summary: generic linux kernel
description: The generic Ubuntu kernel package as a snap
type: kernel
confinement: strict
build-base: devel

architectures:
  - amd64
  - arm64

parts:
  kernel:
    source: https://git.launchpad.net/~canonical-kernel-snaps/+git/kernel-snaps-uc24
    source-type: git
    source-branch: main
    plugin: nil
    override-pull: |
      snapcraftctl pull
      snapcraftctl set-version "$(apt show linux-image-uc-generic 2>/dev/null | sed -n 's/^Version: //p' | head -n1)"
    build-packages:
      - kmod
    stage-packages:
      - linux-image-uc-generic
    organize:
      boot: ./
      lib/firmware: firmware
      lib/modules: modules
      usr/share/doc: doc
    override-build: |
      mv "$SNAPCRAFT_PART_INSTALL"/boot/kernel.efi-* "$SNAPCRAFT_PART_INSTALL"/boot/kernel.efi
      # Update depmod
      depmod -b "$SNAPCRAFT_PART_INSTALL" $(basename "$SNAPCRAFT_PART_INSTALL"/lib/modules/*)
      # Trim firmware
      "$SNAPCRAFT_PART_SRC"/trim-firmware "$SNAPCRAFT_PART_INSTALL"/lib
      snapcraftctl build
```

### Relevant log output

```shell
Bad snapcraft.yaml content:
- grade must be 'devel' when build-base is 'devel'                                                                                                            
Full execution log: '/home/xnox/.local/state/snapcraft/log/snapcraft-20230717-222251.915806.log'            

Traceback (most recent call last):
  File "/snap/snapcraft/9362/bin/snapcraft", line 8, in <module>
    sys.exit(run())
  File "/snap/snapcraft/9362/lib/python3.8/site-packages/snapcraft/cli.py", line 255, in run
    _run_dispatcher(dispatcher, global_args)
  File "/snap/snapcraft/9362/lib/python3.8/site-packages/snapcraft/cli.py", line 228, in _run_dispatcher
    dispatcher.run()
  File "/snap/snapcraft/9362/lib/python3.8/site-packages/craft_cli/dispatcher.py", line 448, in run
    return self._loaded_command.run(self._parsed_command_args)
  File "/snap/snapcraft/9362/lib/python3.8/site-packages/snapcraft/commands/lifecycle.py", line 265, in run
    super().run(parsed_args)
  File "/snap/snapcraft/9362/lib/python3.8/site-packages/snapcraft/commands/lifecycle.py", line 138, in run
    parts_lifecycle.run(self.name, parsed_args)
  File "/snap/snapcraft/9362/lib/python3.8/site-packages/snapcraft/parts/lifecycle.py", line 216, in run
    _run_command(
  File "/snap/snapcraft/9362/lib/python3.8/site-packages/snapcraft/parts/lifecycle.py", line 262, in _run_command
    _run_in_provider(project, command_name, parsed_args)
  File "/snap/snapcraft/9362/lib/python3.8/site-packages/snapcraft/parts/lifecycle.py", line 537, in _run_in_provider
    build_base = providers.SNAPCRAFT_BASE_TO_PROVIDER_BASE[snapcraft_base]
KeyError: 'core24'
```

**Comment by sergiusens** [comment]
This is on purpose, since there is no stable core24 yet. Is this only for kernels, it could make sense to be able to override there. @pedronis what are your thoughts on this?

**Comment by pedronis** [comment]
why does 24-hwe/stable exists already?  we cannot be sure to we have a definitive initrd API for it at this time?

anyway from the point of core24 usage to build a kernel, I suspect the dependence between a kernel and the base is looser than for an app, so from that POV that is probably something we could override

**Comment by xnox** [comment]
Given some teams are trying to deliver products on top of 23.10 and will have 24.04-devel in parallel I have now opened kernel 23.10 track, set build-base to core22, yet override launchpad recipe to build on mantic. Which is impossible to reproduce locally.

Ideally, I would want to use build-base: devel, and publish kernel snaps based on mantic kernel & mantic initrd-userspace to 23.10 track.

And yes 23.10/stable will exist => the snap is signed with production secureboot keys, and goes onto images that do track stable out of futuristic tracks.

stable risk must exist, as nobody wants to build even daily images out of candidate risk => since users that install those systems will never have ability to switch to stable risk (no remodel support). And reinstalling human facing machines is pain.

**Comment by xnox** [comment]
I am now unable to build noble v6.6 kernel snap, both locally and in launchpad, using the same recipe, which i must do during noble development to promote both v6.6 base kernel snap & deb all the way to 24/stable for building and testing Ubuntu Desktop iso which uses both kernel in .deb and snap form.

When will snapcraft be available for use that support `build-base:core24` for snaps of type:kernel?

**Comment by mr-cal** [comment]
Please re-open this issue if I'm wrong, but I believe this is no longer relevant (building a core24 kernel snap prior to the release of core24).
